src/main/java/org/prebid/server/deals/DealsProcessor.java

```diff
@@ -129,34 +128,27 @@ public DealsProcessor(LineItemService lineItemService,
         final Promise<Tuple3<DeviceInfo, GeoInfo, UserDetails>> promise = Promise.promise();
         compositeFuture.onComplete(ignored -> handleDealsInfo(compositeFuture, promise, context.getAccount().getId()));
         return promise.future()
-                .map((Tuple3<DeviceInfo, GeoInfo, UserDetails> tuple) ->
-                        enrichAuctionContext(context, tuple.getLeft(), tuple.getMiddle(), tuple.getRight()))
+                .map(tuple -> enrichAuctionContext(context, tuple))
                 .map(this::matchAndPopulateDeals);
     }
 
-    /**
-     * Finds deals only bidders without matched deals and removes them from {@link Imp}.
-     * If {@link Imp} has only deals only bidders without deals, null will be returned.
-     */
-    public static Imp removeDealsOnlyBiddersWithoutDeals(Imp imp, AuctionContext auctionContext) {
-        final Set<String> dealsOnlyBiddersToRemove = getDealsOnlyBiddersToRemove(imp);
-        Imp resolvedImp = imp;
-        if (CollectionUtils.isNotEmpty(dealsOnlyBiddersToRemove)) {
-            final ObjectNode extImp = removeBidders(imp, dealsOnlyBiddersToRemove);
-            resolvedImp = imp.toBuilder().ext(extImp).build();
-            if (hasBidder(resolvedImp)) {
-                logBidderOrImpExclusion(auctionContext, imp, dealsOnlyBiddersToRemove, "No Line Items from "
-                        + "bidders %s matching imp with id %s and ready to serve. Removing impression from request"
-                        + " to these bidders because dealsonly flag is on for them.");
-            } else {
-                logBidderOrImpExclusion(auctionContext, imp, dealsOnlyBiddersToRemove, "No Line Items from "
-                        + "bidders %s matching imp with id %s and ready to serve. Removing imp from requests to all"
-                        + " bidders because dealsonly flag is on for these bidders and no other valid bidders in imp.");
-                resolvedImp = null;
+    public static Imp removePgDealsOnlyBiddersWithoutDeals(AuctionContext auctionContext,
```

Comment:
This class already has mapper as class variable, so let's remove static access from method, and use class defined mapper, also pls check similar methods